### PR TITLE
Feat/implement solana staking

### PR DESCRIPTION
**Pull request by martykan**
## Description

Local version of #15409 rebased on current develop

Only major change from me is 98175b55ff53d4b89632bfc526e6401161f93385, which solves build issues of Solana worker due to cross-fetch. 

Other changes are mainly formatting and lint + fixup squashing

### CODE REVIEW

File: packages/blockchain-link-utils/src/blockbook.ts [hunk 1]
```diff
@@ -87,21 +89,15 @@ export const filterTokenTransfers = (
...
-    const { poolInstance, withdrawTreasury } = ethereumStakingAddresses;
+    const poolInstances = Object.values(ETH_NETWORK_ADDRESSES).map(
+        network => network.addressContractPool,
+    );
+    const withdrawTreasuries = Object.values(ETH_NETWORK_ADDRESSES).map(
+        network => network.addressContractWithdrawTreasury,
+    );
 
-    return poolInstance.includes(from) && withdrawTreasury.includes(to);
+    return poolInstances.includes(from) && withdrawTreasuries.includes(to);
```
Comment (mroz22) [inline packages/blockchain-link-utils/src/blockbook.ts:100]: is this related? or just some general refactoring?

File: packages/suite/package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
         "@sentry/core": "^7.100.1",
         "@solana/buffer-layout": "^4.0.1",
         "@solana/web3.js": "^2.0.0",
+        "@solana/web3.js-version1": "npm:@solana/web3.js@1.95.8",
```
Comment (mroz22) [inline packages/suite/package.json:30]: this looks like a todo

File: packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewOutputList/TransactionReviewOutputList.tsx [hunk 3]
```diff
@@ -99,7 +99,7 @@ export interface TransactionReviewOutputListProps {
     actionText: TranslationKey;
     isSending?: boolean;
     setIsSending?: () => void;
-    ethereumStakeType?: StakeType;
+    stakeType?: StakeType;
```
Comment (mroz22) [inline packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewOutputList/TransactionReviewOutputList.tsx:102]: I can see bunch of refactoring looking changes across  this PR. don't you guys want to review them separately?

File: packages/suite/package.json (hunk 2 above)
Comment (tomasklim) [reply packages/suite/package.json:30]: yea, Everstake has to migrate to v2 https://satoshilabs.slack.com/archives/C0543DJBK0C/p1733907109990759

File: packages/blockchain-link-utils/src/blockbook.ts (hunk 1 above)
Comment (tomasklim) [reply packages/blockchain-link-utils/src/blockbook.ts:100]: the previous version of everstake sdk was not working in blockchain-link. Now after update, it does. So we were able to get rid of hardcoded addresses which were used just for filtering for tx history

**Comment by martykan** [comment]
@SocketSecurity ignore-all

File: packages/blockchain-link-utils/src/blockbook.ts (hunk 1 above)
Comment (mroz22) [reply packages/blockchain-link-utils/src/blockbook.ts:100]: ok

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx [hunk 4]
```diff
@@ -29,7 +29,12 @@ export const StakeModal = ({ onCancel }: StakeModalModalProps) => {
         <StakeEthFormContext.Provider value={stakeEthContextValues}>
             <NewModal
                 size="huge"
-                heading={<Translation id="TR_STAKE_ETH" />}
+                heading={
+                    <Translation
+                        id="TR_STAKE_NETWORK"
+                        values={{ networkSymbol: account.symbol.toUpperCase() }}
```
Comment (tomasklim) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx]:
```suggestion
                        values={{ symbol: account.symbol.toUpperCase() }}
```
please, otherwise @adderpositive will miss it

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/EverstakeModal.tsx [hunk 5]
```diff
@@ -22,9 +22,51 @@ export const EverstakeModal = ({ onCancel }: EverstakeModalProps) => {
...
+            icon: 'fileFilled',
+            message: (
+                <Translation
+                    id={
+                        account?.networkType === 'ethereum'
+                            ? 'TR_STAKE_EVERSTAKE_MANAGES'
+                            : 'TR_STAKE_BY_STAKING_YOU_CAN_EARN_REWARDS'
+                    }
+                    values={{
+                        networkSymbol: account?.symbol.toUpperCase(),
```
Comment (tomasklim) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/EverstakeModal.tsx]: @adderpositive is it ok here or not?

Comment (adderpositive) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/EverstakeModal.tsx]: Yes, it is ok

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx (hunk 4 above)
Comment (adderpositive) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx]: This one is also ok. It is for a translation.

File: suite-common/wallet-utils/src/solanaStakingUtils.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import { SolanaStakingAccount } from '@trezor/blockchain-link-types/src/solana';
+import {
+    Account,
+    supportedSolanaNetworkSymbols,
+    SupportedSolanaNetworkSymbols,
+} from '@suite-common/wallet-types';
+import { LAMPORTS_PER_SOL } from '@suite-common/wallet-constants';
+
+export function isSupportedSolStakingNetworkSymbol(
+    networkSymbol: NetworkSymbol,
```
Comment (adderpositive) [inline suite-common/wallet-utils/src/solanaStakingUtils.ts]: Use please `symbol` instead of `networkSymbol`

File: suite-common/wallet-utils/src/solanaStakingUtils.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+    return new BigNumber(amount).div(LAMPORTS_PER_SOL).toFixed(9);
+};
+
+interface SolNetworkConfig {
+    network: SolNetwork;
+}
+
+export const getSolNetworkForWalletSdk = (symbol: NetworkSymbol): SolNetworkConfig => {
+    const solNetworks: { [key in NetworkSymbol]?: SolNetworkConfig } = {
```
Comment (adderpositive) [inline suite-common/wallet-utils/src/solanaStakingUtils.ts]: Consider using `PartialRecord<NetworkSymbol, SolNetworkConfig>`

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx (hunk 4 above)
Comment (adderpositive) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx]: I see. If the translation is new, please consider using `symbol` as @tomasklim suggested.

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/EverstakeModal.tsx (hunk 5 above)
Comment (adderpositive) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/EverstakeModal.tsx]: If it is a new translation, `symbol` would be better. )

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx (hunk 4 above)
Comment (martykan) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx]: Ok, I will unify the translations to use `symbol`. Most of the messages are new/changed

File: suite-common/wallet-utils/src/solanaStakingUtils.ts (hunk 6 above)
Comment (martykan) [reply suite-common/wallet-utils/src/solanaStakingUtils.ts]: Is this important? This is not for display but actually based on network

Comment (martykan) [reply suite-common/wallet-utils/src/solanaStakingUtils.ts]: Or you know what, I'll just change it 😃

**Review by tomasklim — APPROVED** [review]
LGTM
Three small text bugs in the staking flow
Screenshot 2024-12-17 at 21 35 29
Screenshot 2024-12-17 at 21 35 20
Screenshot 2024-12-17 at 21 34 58

File: suite-common/wallet-utils/src/solanaStakingUtils.ts (hunk 6 above)
Comment (adderpositive) [reply suite-common/wallet-utils/src/solanaStakingUtils.ts]: Gracia 🫶

**Comment by dev-pvl** [comment]
@tomasklim I've fixed these bugs but these changes will be in the next PR.

**Comment by bosomt** [comment]
- 🟢 staking section present in account section
- 🟢 so far no info about time estimate `~NaN days` as expected
- 🟢 0.1 minimum enforced
- 🟢  fraction buttons works as expected
- 🔴 staking 0.1 SOL + 0.000025 fee but device says Deposit 0.10228228 
- 🟢 managed to stake [0.10228288 SOL](https://solscan.io/tx/rMayABKegVWFzjTQB72LdwwDZJkjf5H8hXbghCNZijxCF3aSCDW8EaZdKkVjTbJxfm97i5Lq7197cZP94wKr5FB)
- 🔴 Transaction history says unknown transaction with 0.00000503 SOL
Screenshot 2025-01-02 at 14 54 05

- 🟢 transaction detail says 0.10228791 SOL input
Screenshot 2025-01-02 at 14 54 16

**Comment by bosomt** [comment]
**QA OK**

- stake
- unstake
- stake section present in navigation bar
- staking info present in stake tab
- im allowed to unstake any amount so im staking less than 0.1 SOL minimum not sure if expected @tomasklim 
- `Learn more` link navigates to `Stake Ethereum (ETH) in Trezor Suite` that will be probably updated before public release
- both unstake and stake transactions are still reported as `Unknown transaction` in transaction history 

**Info:**
 - Suite version: desktop 25.1.1 (4403882efb2a4a802c3d77917d6350a4fae7663d)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/25.1.1 Chrome/128.0.6613.186 Electron/32.2.6 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3B1 2.8.7 regular (revision f2aeea159f1fde98d4fc5575350a8c9fc4478367)
 - Transport: BridgeTransport 3.0.0-bundled.25.1.1
